utils/build-toolchain

```diff
@@ -0,0 +1,43 @@
+#/bin/bash
+#
+# utils/build-toochain - documents process for building a toolchain
+#
+# From: https://lists.swift.org/pipermail/swift-dev/Week-of-Mon-20151207/000212.html
+#
+# This source file is part of the Swift.org open source project
+#
+# Copyright (c) 2014 - 2016 Apple Inc. and the Swift project authors
+# Licensed under Apache License v2.0 with Runtime Library Exception
+#
+# See http://swift.org/LICENSE.txt for license information
+# See http://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+
+cd "$(dirname $0)/.."
+SRC_DIR=$PWD
+
+YEAR=$(date +"%Y")
+MONTH=$(date +"%m")
+DAY=$(date +"%d")
+TOOLCHAIN_VERSION="swift-SNAPSHOT-${YEAR}-${MONTH}-${DAY}-a"
+ARCHIVE_DIR="${TOOLCHAIN_VERSION}-${BUILD_NUMBER}"
+ARCHIVE="${TOOLCHAIN_VERSION}-osx.tar.gz"
+SYM_ARCHIVE="${TOOLCHAIN_VERSION}-osx-symbols.tar.gz"
+BUNDLE_IDENTIFIER="org.swift.${YEAR}${MONTH}${DAY}"
+DISPLAY_NAME="Local Swift Development Snapshot ${YEAR}-${MONTH}-${DAY}"
+TOOLCHAIN_NAME="${TOOLCHAIN_VERSION}"
+
+SWIFT_SOURCE_ROOT="${SRC_DIR}"
+SWIFT_BUILD_ROOT="${SRC_DIR}/build"
+SWIFT_INSTALLABLE_PACKAGE="${SRC_DIR}/${ARCHIVE}"
+SWIFT_INSTALL_DIR="${SRC_DIR}/swift-nightly-install"
+SWIFT_INSTALL_SYMROOT="${SRC_DIR}/swift-nightly-symroot"
+SWIFT_TOOLCHAIN_DIR="/Library/Developer/Toolchains/${TOOLCHAIN_NAME}.xctoolchain"
+SYMBOLS_PACKAGE="${SRC_DIR}/${SYM_ARCHIVE}"
+
+if [ -e /etc/lsb-release ]; then
```

Comment:
Is this to determine the host system? If so, `util/build-script-impl` uses `uname -s` for this purpose.